### PR TITLE
Search

### PR DESCRIPTION
**Pull request by sadafpartovi**
Added color options in global css and context.

### CODE REVIEW

File: components/navbar/Nav.jsx [hunk 1]
```diff
@@ -1,50 +1,80 @@
...
-      </Link>
-      }
-      {!auth.currentUser ?
-        <i onClick={() => showModal()}className={`fa fa-user fa-lg ${style.icon}`}/>
-      : <Link href='/profile'>
-      <i className={`fa fa-user fa-lg ${style.icon}`}/>
-      </Link>
-      }
-    </div>
+      <div className={styles.modalContainer}>
```
Comment (robin-son1) [inline components/navbar/Nav.jsx:38]: nice refactoring of the login check

File: context/Provider.jsx [hunk 2]
```diff
@@ -14,25 +14,36 @@ const Provider = ({children})=>{
...
+    'Arts & Crafts': 'fa-paint-brush',
+    'Video Games': 'fa-gamepad',
+  })
 
+  const [colorSchema, setColorSchema] = useState({
+    'Orange ': '#FF6E40',
+    'Light Gray': '#EBEBEB',
+    'Dark Gray': ' #484848',
+    'Navy Blue': '#1E3D59',
+    'White': '#fff',
```
Comment (matt-heindel) [inline context/Provider.jsx:31]: Thanks for adding these! Will be nice to have consistency across App

File: components/navbar/Modal.jsx [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
-      <Button onClick={onOpen}>Trigger modal</Button>
       <Modal onClose={onClose} isOpen={isOpen} isCentered>
         <ModalOverlay />
         <ModalContent>
-          <ModalHeader>Modal Title</ModalHeader>
+          <ModalHeader>oops, You Need to Login to Access</ModalHeader>
           <ModalCloseButton />
           <ModalBody>
-            <Lorem count={2} />
+           <Button>Login</Button>
```
Comment (matt-heindel) [inline components/navbar/Modal.jsx:21]: Does this button navigate somewhere? In my testing it did not

File: components/search/SearchBar.jsx [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { Flex, Spacer, InputGroup, InputLeftElement, Input,SearchIcon , PhoneIco
 import MainContext from '../../context/MainContext';
 import {useContext, useState} from 'react';
 
+import styles from '../../styles/SearchNav.module.css';
+
```
Comment (matt-heindel) [inline components/search/SearchBar.jsx:7]: Nice way to handle styles

File: context/Provider.jsx [hunk 5]
```diff
@@ -14,25 +14,36 @@ const Provider = ({children})=>{
...
   })
 
   const searchData = async() => {
     let {data} = await axios.get('/api/getPosts')
     setExploreData(data)
     setFilterData(data)
     setCategory([...new Set(data.map(item => item.category))])
-    console.log(category)
```
Comment (matt-heindel) [inline context/Provider.jsx:27]: Thanks for removing these console logs

File: pages/search/index.js [hunk 6]
```diff
@@ -7,10 +7,11 @@ const MenuBar = dynamic(() => import('../../components/search/MenuBar'), {
 import SearchPost from '../../components/search/SearchPost';
 import Nav from '../../components/navbar/Nav';
 
+
 const Search = () => {
   return (
     <>
-      <div style={{position: 'sticky', top: '0', backgroundColor: 'white'}}>
+      <div style={{position: 'sticky', top: '0', backgroundColor: '#1E3D59'}}>
```
Comment (matt-heindel) [inline pages/search/index.js:14]: Should this color come from Context instead of being hard coded?

File: styles/SearchMenu.module.css [hunk 7]
```diff
@@ -1,9 +1,9 @@
...
-  margin: 5px;
-  min-width: 100px;
+  padding: 12px;
+  background-color: #EBEBEB;
+  margin: 10px;
+  min-width: 160px;
   border-radius: 10px;
   text-align: center;
-  border: 1px solid gray;
+  color: #484848;
```
Comment (matt-heindel) [inline styles/SearchMenu.module.css:8]: Should these colors come from Context instead of being hard coded?

File: components/navbar/Modal.jsx [hunk 8]
```diff
@@ -1,23 +1,24 @@
...
+
   Modal, ModalOverlay, ModalContent, ModalHeader, ModalFooter, ModalBody, ModalCloseButton,
   Input, Button, useDisclosure, FormControl, FormLabel, Textarea, Lorem
 } from "@chakra-ui/react";
+import Login from '../login/Login';
 
-const MyModal = () =>  {
-  const { isOpen, onOpen, onClose } = useDisclosure()
+const MyModal = ({isOpen, onClose}) =>  {
+  // const { isOpen, onOpen, onClose } = useDisclosure()
```
Comment (matt-heindel) [inline components/navbar/Modal.jsx:11]: cruft?

**Review by matt-heindel — APPROVED** [review]
Nice work @sadafpartovi 

I left a couple of comments for your to check out

One thing I noticed is that if I filter by a topic with only a few posts, the background color changes to white, see below

image

**Pull request merged**